src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,71 @@
+const expensesServices = require('../services/expenses/expenses.services');
+const { isExpenseValid } = require('../untils/expenses');
+const { isUserExist } = require('../untils/users');
+
+const getExpenses = (req, res) => {
+  const { categories, userId, from, to } = req.query;
+
+  return res
+    .status(200)
+    .json(expensesServices.getExpenses(categories, userId, from, to));
+};
+
+const getExpenseById = (req, res) => {
+  const { id } = req.params;
+
+  const expenseId = Number(id);
+
+  if (!expenseId || isNaN(expenseId)) {
+    res.sendStatus(400);
+  }
+
+  const expense = expensesServices.getExpenseById(expenseId);
+
+  return !expense ? res.sendStatus(404) : res.status(200).json(expense);
+};
+
+const saveExpense = (req, res) => {
+  if (!isExpenseValid(req.body) || !isUserExist(+req.body.userId)) {
```

Comment:
Ensure that `isExpenseValid` and `isUserExist` functions are correctly implemented to handle all necessary validations. If these functions throw errors or return unexpected results, it could lead to incorrect status codes being sent.